app/services/service_provider_logo_updater.rb

```diff
@@ -0,0 +1,123 @@
+require 'login_gov/hostdata'
+require 'subprocess'
+
+class ServiceProviderLogoUpdater
+  include ServiceProviderHelper
+
+  # :reek:TooManyStatements
+  # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+  def import_logos_to_active_storage
+    return unless Figaro.env.logo_upload_enabled
+
+    git_latest_idp_config
+    idp_config.each do |sp|
+      issuer = sp['issuer']
+      logo_name = sp['logo']
+      logger.info('~~ ' + issuer.to_s + ' logo: ' + logo_name.to_s)
+      next unless logo_name.present? && valid_image_type?(logo_name)
+
+      service_provider = ServiceProvider.find_by(issuer: issuer)
+      next unless service_provider
+
+      service_provider.logo_file.attach(
+        io: File.open(logo_path(logo_name)),
+        filename: logo_name,
+        content_type: mime_type(logo_name)
+      )
+    end
+  rescue StandardError => error
+    handle_error(error)
+  end
+  # rubocop:enable Metrics/AbcSize, Metrics/MethodLength
+
+  #############################################################################
+
+  # Clone the private-but-not-secret git repo
+  def git_latest_idp_config
+    if File.directory? repo_dir
+      update_repo
+    else
+      clone_repo
+    end
+  end
+
+  def idp_config
+    @idp_config ||= load_idp_config
+  end
+
+  def handle_error(error)
+    # ::NewRelic::Agent.notice_error(error)
+    logger.error(error)
+  end
+
+  # :reek:UtilityFunction
+  def logo_path(filename)
+    Rails.root.join(
+      'identity-idp-config',
+      'public',
+      'assets',
+      'images',
+      'sp-logos',
+      filename
+    )
+  end
+
+  #############################################################################
+
+  def update_repo
+    cmd = ['cd', repo_dir, ';', 'git', 'pull']
```

Comment:
I don't like `cd` without changing back...what if we did:

```
Dir.chdir(repo_dir) do
  # git pull ... etc
end
```

Reply:
Unsure if subprocess resets the working dir after commands, but I think this form is clearer

Reply:
Good call.